### PR TITLE
Simplify Handler registration.

### PR DESCRIPTION
**Pull request by cschuet**

### CODE REVIEW

File: cartographer_grpc/framework/rpc_handler_interface.h [hunk 1]
```diff
@@ -28,6 +28,9 @@ class Rpc;
 class RpcHandlerInterface {
  public:
   virtual ~RpcHandlerInterface() = default;
+  // Returns the fully qualified name of the gRPC method this handler is
+  // implementing.
+  virtual std::string method_name() const = 0;
```
Comment (gaschler) [inline cartographer_grpc/framework/rpc_handler_interface.h]: Perhaps add a comment where one can look this up when implementing a new handler?

File: cartographer_grpc/framework/server.h [hunk 2]
```diff
@@ -55,12 +55,15 @@ class Server {
...
-                           << method_name;
-      rpc_handlers_[ServiceType::service_full_name()].emplace(
+    template <typename RpcHandlerType>
+    void RegisterHandler() {
+      std::string method_full_name = GetMethodFullName<RpcHandlerType>();
+      std::string service_full_name;
+      std::string method_name;
+      std::tie(service_full_name, method_name) =
+          ParseMethodFullName(method_full_name);
+
```
Comment (gaschler) [inline cartographer_grpc/framework/server.h]: can you check that ServiceType::service_full_name() == service_full_name?

File: cartographer_grpc/framework/server.h [hunk 3]
```diff
@@ -74,12 +77,20 @@ class Server {
...
 
    private:
     using ServiceInfo = std::map<std::string, RpcHandlerInfo>;
 
+    template <typename RpcHandlerType>
+    std::string GetMethodFullName() {
+      auto handler = cartographer::common::make_unique<const RpcHandlerType>();
+      return handler->method_name();
+    }
+    std::tuple<std::string, std::string> ParseMethodFullName(
```
Comment (gaschler) [inline cartographer_grpc/framework/server.h]: std::string /* description */

File: cartographer_grpc/framework/server_test.cc [hunk 4]
```diff
@@ -150,6 +166,13 @@ class ServerTest : public ::testing::Test {
 };
 
 TEST_F(ServerTest, StartAndStopServerTest) {
+  auto* pool = google::protobuf::DescriptorPool::generated_pool();
+  CHECK(pool->FindMessageTypeByName(
```
Comment (gaschler) [inline cartographer_grpc/framework/server_test.cc]: EXPECT_NOT_NULL instead of CHECK

File: cartographer_grpc/framework/server.h (hunk 2 above)
Comment (gaschler) [inline cartographer_grpc/framework/server.h]: drop empty line

File: cartographer_grpc/framework/rpc_handler_interface.h (hunk 1 above)
Comment (cschuet) [reply cartographer_grpc/framework/rpc_handler_interface.h]: Done.

File: cartographer_grpc/framework/server.h (hunk 2 above)
Comment (cschuet) [reply cartographer_grpc/framework/server.h]: Done.

Comment (cschuet) [reply cartographer_grpc/framework/server.h]: Unfortunately not.

File: cartographer_grpc/framework/server.h (hunk 3 above)
Comment (cschuet) [reply cartographer_grpc/framework/server.h]: Done.

File: cartographer_grpc/framework/server_test.cc (hunk 4 above)
Comment (cschuet) [reply cartographer_grpc/framework/server_test.cc]: Sorry this was debug stuff. Removed.

**Comment by cschuet** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user cschuet. Merge queue now has a length of 1.

**Pull request merged**